### ISSUE TITLE
There are several accessibility issues with devdocs

### ISSUE THREAD

**Issue by ShaneQful** [Thibaut/devdocs#305]
There are several accessibility issues with devdocs which may impair users of assistive technologies such as screen reader users from using devdocs effectively:

* <s>Violations of WCAG colour contrast limits</s>
* There are several areas which could use aria roles could be used to improve the screenreader and keyboard usuability of the site
* Aria live regions would be good to allow screen readers know the documentation pane has updated 

I'd like to have a look at some of the issues above, just a heads up in case other people are working on the same

**Comment by Thibaut** [comment]
Hey!

Good to know. There's definitely room for improvement.

> Violations of WCAG colour contrast limits

This is a tricky problem as I find that high contrast UIs don't look great on high quality displays (which already have good contrast). To get the best of both worlds, maybe we could create a separate high-contrast /  "accessibility" color scheme?

**Comment by ShaneQful** [comment]
I'll probably fix the issues with it in high contrast mode first but the WCAG guidelines are supposed to be followed regardless of the mode. That said most of the application complies with the 4.5 minimum contrast.

**Comment by ShaneQful** [comment]
@Thibaut is there a particular reason there is a meta tag which disables scaling, didn't want to just remove it as I presumed it was put there to solve a particular problem.

`<meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1,user-scalable=no,shrink-to-fit=no">`

**Comment by Thibaut** [comment]
> is there a particular reason there is a meta tag which disables scaling?

The UI breaks when zoomed:

![iphone 5s - iphone 5s ios 9 2 13c75 2016-02-07 09-25-05](https://cloud.githubusercontent.com/assets/17579/12872877/2babef0c-cd7d-11e5-8dde-25aa98880a3e.png) ![iphone 5s - iphone 5s ios 9 2 13c75 2016-02-07 09-25-17](https://cloud.githubusercontent.com/assets/17579/12872879/2e9c7100-cd7d-11e5-8ee6-221ed250c618.png)


`user-scalable=no` also removes the 300ms tap/click delay on [Android](https://developers.google.com/web/updates/2013/12/300ms-tap-delay-gone-away) and soon [iOS](https://trac.webkit.org/changeset/191072).

**Comment by ShaneQful** [comment]
It's technically a WCAG failure but considering all the desktop browsers ignore it for the most part and it's very useful for mobile there's no great point in removing it.

**Comment by Thibaut** [comment]
Added some aria role attributes and cleaned up the markup here: https://github.com/Thibaut/devdocs/commit/dd86527da5392e38c76832dd27010e186b0095a5

More than that (live regions, etc.) would require extensive testing. Not something I'm interested in doing at the moment (sorry), so closing this issue. If anyone is interested in pushing this forward, PRs are more than welcome.
